channels/rdpgfx/server/rdpgfx_main.c

```diff
@@ -0,0 +1,1659 @@
+
+	for (index = 0; index < pdu->monitorCount; index++)
+	{
+		monitor = &(pdu->monitorDefArray[index]);
+		Stream_Write_UINT32(s, monitor->left); /* left (4 bytes) */
+		Stream_Write_UINT32(s, monitor->top); /* top (4 bytes) */
+		Stream_Write_UINT32(s, monitor->right); /* right (4 bytes) */
+		Stream_Write_UINT32(s, monitor->bottom); /* bottom (4 bytes) */
+		Stream_Write_UINT32(s, monitor->flags); /* flags (4 bytes) */
+	}
+
+	/* pad (total size must be 340 bytes) */
+	Stream_SetPosition(s, RDPGFX_RESET_GRAPHICS_PDU_SIZE);
+	return rdpgfx_server_single_packet_send(context, s);
+}
+
+/**
+ * Function description
+ *
+ * @return 0 on success, otherwise a Win32 error code
+ */
+static UINT rdpgfx_send_evict_cache_entry_pdu(RdpgfxServerContext* context,
+		RDPGFX_EVICT_CACHE_ENTRY_PDU* pdu)
+{
+	wStream* s = rdpgfx_server_single_packet_new(
+					 RDPGFX_CMDID_EVICTCACHEENTRY,
+					 sizeof(RDPGFX_EVICT_CACHE_ENTRY_PDU));
+
+	if (!s)
+	{
+		WLog_ERR(TAG, "rdpgfx_server_single_packet_new failed!");
+		return CHANNEL_RC_NO_MEMORY;
+	}
+
+	Stream_Write_UINT16(s, pdu->cacheSlot); /* cacheSlot (2 bytes) */
+	return rdpgfx_server_single_packet_send(context, s);
+}
+
+/**
+ * Function description
+ *
+ * @return 0 on success, otherwise a Win32 error code
+ */
+static UINT rdpgfx_send_cache_import_reply_pdu(RdpgfxServerContext* context,
+		RDPGFX_CACHE_IMPORT_REPLY_PDU* pdu)
+{
+	UINT16 index;
+	wStream* s = rdpgfx_server_single_packet_new(
+					 RDPGFX_CMDID_CACHEIMPORTREPLY,
+					 sizeof(RDPGFX_CACHE_IMPORT_REPLY_PDU) +
+					 sizeof(UINT16) * pdu->importedEntriesCount);
+
+	if (!s)
+	{
+		WLog_ERR(TAG, "rdpgfx_server_single_packet_new failed!");
+		return CHANNEL_RC_NO_MEMORY;
+	}
+
+	/* importedEntriesCount (2 bytes) */
+	Stream_Write_UINT16(s, pdu->importedEntriesCount);
+
+	for (index = 0; index < pdu->importedEntriesCount; index++)
+	{
+		Stream_Write_UINT16(s, pdu->cacheSlots[index]); /* cacheSlot (2 bytes) */
+	}
+
+	return rdpgfx_server_single_packet_send(context, s);
+}
+
+/**
+ * Function description
+ *
+ * @return 0 on success, otherwise a Win32 error code
+ */
+static UINT rdpgfx_send_create_surface_pdu(RdpgfxServerContext* context,
+		RDPGFX_CREATE_SURFACE_PDU* pdu)
+{
+	wStream* s = rdpgfx_server_single_packet_new(
+					 RDPGFX_CMDID_CREATESURFACE,
+					 sizeof(RDPGFX_CREATE_SURFACE_PDU));
+
+	if (!s)
+	{
+		WLog_ERR(TAG, "rdpgfx_server_single_packet_new failed!");
+		return CHANNEL_RC_NO_MEMORY;
+	}
+
+	Stream_Write_UINT16(s, pdu->surfaceId); /* surfaceId (2 bytes) */
+	Stream_Write_UINT16(s, pdu->width); /* width (2 bytes) */
+	Stream_Write_UINT16(s, pdu->height); /* height (2 bytes) */
+	Stream_Write_UINT8(s, pdu->pixelFormat); /* RDPGFX_PIXELFORMAT (1 byte) */
+	return rdpgfx_server_single_packet_send(context, s);
+}
+
+/**
+ * Function description
+ *
+ * @return 0 on success, otherwise a Win32 error code
+ */
+UINT rdpgfx_send_delete_surface_pdu(RdpgfxServerContext* context,
+									RDPGFX_DELETE_SURFACE_PDU* pdu)
```

Comment:
Because github tab = 8 space, while our non-documented code style define tab=4 space, see
cat scripts/format_code.sh
#!/bin/bash
<snip>
$ASTYLE --lineend=linux --mode=c --indent=force-tab=4 --brackets=linux --pad-header \
                                                           --indent-switches --indent-cases --indent-preprocessor \
                                                           --indent-col1-comments --delete-empty-lines --break-closing-brackets \
                                                           --align-pointer=type --indent-labels --brackets=break \
                                                           --unpad-paren --break-blocks $@
                                                           exit $?
....